src/data/storage.ts

```diff
@@ -0,0 +1,127 @@
+import * as idb from "idb-keyval";
+import { ExcalidrawElement } from "../element/types";
+import { AppState } from "../types";
+import { clearAppStateForLocalStorage } from "../appState";
+import { restore } from "./restore";
+
+const STORAGE_KEY = "excalidraw";
+const STORAGE_KEY_STATE = "excalidraw-state";
+const STORAGE_KEY_COLLAB = "excalidraw-collab";
+
+/**
+ * Uses indexedDB as key-value storage.
+ * Fallsback to localStorage on older browsers without IDB (or) while running tests.
+ */
+export class WebStorageProvider {
+  private supportsIDB: boolean;
+  private indexedDBStore: idb.Store | null;
+
+  constructor() {
+    this.supportsIDB = "indexedDB" in window && process.env.NODE_ENV !== "test";
```

Comment:
This disables using IDB for tests. Although all the tests pass, while running it with IDB. We can remove this check, but I found that most tests have a `localStorage.clear()` in them in the `beforeEach` block.